Drop AtomString constructors that take in a raw pointer and a length

cdumez opened the pull request:
#### 36f7876c579ec5bc800b2e01934a012ca35a6444
<pre>
Drop AtomString constructors that take in a raw pointer and a length
<a href="https://bugs.webkit.org/show_bug.cgi?id=272132">https://bugs.webkit.org/show_bug.cgi?id=272132</a>

Reviewed by Brent Fulgham.

Drop AtomString constructors that take in a raw pointer and a length, in favor
of the ones taking in a std::span.

* Source/JavaScriptCore/runtime/IntlDisplayNames.cpp:
(JSC::IntlDisplayNames::of const):
* Source/JavaScriptCore/runtime/JSString.cpp:
(JSC::JSRopeString::resolveRopeToAtomString const):
* Source/WTF/wtf/text/AtomString.cpp:
(WTF::AtomString::convertASCIICase const):
* Source/WTF/wtf/text/AtomString.h:
* Source/WTF/wtf/text/StringConcatenate.h:
(WTF::tryMakeAtomStringFromAdapters):
* Source/WTF/wtf/text/StringView.cpp:
(WTF::convertASCIILowercaseAtom):
* Source/WTF/wtf/text/StringView.h:
(WTF::StringView::toAtomString const):
* Source/WebCore/bindings/js/SerializedScriptValue.cpp:
(WebCore::CloneDeserializer::readString):
* Source/WebCore/dom/DatasetDOMStringMap.cpp:
(WebCore::convertPropertyNameToAttributeName):
* Source/WebCore/dom/SpaceSplitString.cpp:
(WebCore::TokenAtomStringInitializer::processToken):
* Source/WebCore/html/parser/AtomHTMLToken.h:
(WebCore::AtomHTMLToken::AtomHTMLToken):
* Source/WebCore/html/parser/HTMLNameCache.h:
(WebCore::HTMLNameCache::makeAtomString):
(WebCore::HTMLNameCache::makeQualifiedName):
* Source/WebCore/layout/formattingContexts/inline/text/TextUtil.cpp:
(WebCore::Layout::TextUtil::ellipsisTextRun):
* Source/WebCore/loader/appcache/ApplicationCacheStorage.cpp:
(WebCore::parseHeader):
(WebCore::parseHeaders):
* Source/WebCore/page/EventSource.cpp:
(WebCore::EventSource::parseEventStreamLine):
* Source/WebCore/platform/graphics/FontCache.cpp:
(WebCore::FontCache::useBackslashAsYenSignForFamily):
* Source/WebCore/rendering/style/RenderStyle.cpp:
(WebCore::RenderStyle::hyphenString const):
(WebCore::RenderStyle::textEmphasisMarkString const):

Canonical link: <a href="https://commits.webkit.org/277064@main">https://commits.webkit.org/277064@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9261c29f816a149f299bf9cc831b577721d700b5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/46544 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/25701 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/49150 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/49221 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/42585 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/48851 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/30061 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/23163 "Built successfully") | [⏳ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/WPE-WK2-Tests-EWS "Waiting to run tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/47123 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/22704 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/40122 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/19220 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/20072 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/41236 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/4589 "Built successfully") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/39791 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/42875 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/41615 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/51064 "Built successfully") | 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/46027 "Build is in progress. Recent messages:OS: Monterey (12.6.8), Xcode: 13.4.1; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running compile-jsc") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/21550 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/17970 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/45214 "Passed tests") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/22841 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/44156 "Passed tests") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/23271 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/53169 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6509 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/22544 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/10906 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WTF/wtf/text/StringView.h:
```diff
@@ -659,8 +659,8 @@ inline String StringView::toString() const
-        return AtomString(characters8(), m_length);
-    return AtomString(characters16(), m_length);
+        return span8();
+    return span16();
```
If I were Darin I'd say you should do:

`return is8Bit() ? span8() : span16();`

:-)

cdumez replied:
I don't think he would when the types returned by each side of the ternary are different :)

The proposed code wouldn't build, I would have to do:
```
return is8Bit() ? AtomString(span8()) : AtomString(span16());
```

which is not as nice. Overall, I prefer my version.

brentfulgham replied:
touché! Please do keep your version,

brentfulgham commented on Source/WebCore/loader/appcache/ApplicationCacheStorage.cpp:
```diff
@@ -1054,15 +1054,15 @@ bool ApplicationCacheStorage::storeNewestCache(ApplicationCacheGroup& group)
-    ASSERT(WTF::find(header, headerLength, ':') != notFound);
-    unsigned colonPosition = WTF::find(header, headerLength, ':');
+    ASSERT(WTF::find(header.data(), header.size(), ':') != notFound);
+    unsigned colonPosition = WTF::find(header.data(), header.size(), ':');
```
Seems like `WTF::find()` needs a version that takes a `std::span`!

cdumez replied:
Yes, this is planned. I still have a lot of code to port, especially in StringImpl and StringCommon area.

brentfulgham reviewed (approved):
Another nice cleanup! r=me

cdumez committed:
Drop AtomString constructors that take in a raw pointer and a length
https://bugs.webkit.org/show_bug.cgi?id=272132

Reviewed by Brent Fulgham.

Drop AtomString constructors that take in a raw pointer and a length, in favor
of the ones taking in a std::span.

* Source/JavaScriptCore/runtime/IntlDisplayNames.cpp:
(JSC::IntlDisplayNames::of const):
* Source/JavaScriptCore/runtime/JSString.cpp:
(JSC::JSRopeString::resolveRopeToAtomString const):
* Source/WTF/wtf/text/AtomString.cpp:
(WTF::AtomString::convertASCIICase const):
* Source/WTF/wtf/text/AtomString.h:
* Source/WTF/wtf/text/StringConcatenate.h:
(WTF::tryMakeAtomStringFromAdapters):
* Source/WTF/wtf/text/StringView.cpp:
(WTF::convertASCIILowercaseAtom):
* Source/WTF/wtf/text/StringView.h:
(WTF::StringView::toAtomString const):
* Source/WebCore/bindings/js/SerializedScriptValue.cpp:
(WebCore::CloneDeserializer::readString):
* Source/WebCore/dom/DatasetDOMStringMap.cpp:
(WebCore::convertPropertyNameToAttributeName):
* Source/WebCore/dom/SpaceSplitString.cpp:
(WebCore::TokenAtomStringInitializer::processToken):
* Source/WebCore/html/parser/AtomHTMLToken.h:
(WebCore::AtomHTMLToken::AtomHTMLToken):
* Source/WebCore/html/parser/HTMLNameCache.h:
(WebCore::HTMLNameCache::makeAtomString):
(WebCore::HTMLNameCache::makeQualifiedName):
* Source/WebCore/layout/formattingContexts/inline/text/TextUtil.cpp:
(WebCore::Layout::TextUtil::ellipsisTextRun):
* Source/WebCore/loader/appcache/ApplicationCacheStorage.cpp:
(WebCore::parseHeader):
(WebCore::parseHeaders):
* Source/WebCore/page/EventSource.cpp:
(WebCore::EventSource::parseEventStreamLine):
* Source/WebCore/platform/graphics/FontCache.cpp:
(WebCore::FontCache::useBackslashAsYenSignForFamily):
* Source/WebCore/rendering/style/RenderStyle.cpp:
(WebCore::RenderStyle::hyphenString const):
(WebCore::RenderStyle::textEmphasisMarkString const):

Canonical link: https://commits.webkit.org/277064@main
